### ISSUE TITLE
[WP8.1] AuthFlow.InitAuthentication method not respond

### ISSUE THREAD

**Issue by JosephBalzanoDev** [linvi/tweetinvi#268]
Hi everyone,
My app uses Tweetinvi to receive the lastest tweet with specific tags on Twitter because of this they requires an auth. I call, after I created an instance of ConsumerCredentials with AuthKey and SecretKey, the AuthFlow.InitAuthentication for retrive the login url but on Windows Phone 8.1 the response string is empty. On Windows Phone 10 it works properly instead.

Do you have any idea to fix it? :)
Thank you.

**Comment by linvi** [comment]
Looks like a bug. Have you upgraded to latest (0.9.13.0)?

**Comment by JosephBalzanoDev** [comment]
Yes, I have the problem with the lastet version.

**Comment by linvi** [comment]
I will take a look as soon as possible. I have to be honest and admit that I am no longer testing on WP 8.1. The test devices are currently W10, WP10, Raspberry, Linux/Mono.

Thank you for your feedback

**Comment by JosephBalzanoDev** [comment]
Thank you for immediatly responses and for your excellent work :+1:

**Comment by linvi** [comment]
I have tried and it does work properly for me. Could you share your code and some more context/exceptions that could have been thrown. 

Check the [error handling documentation](https://github.com/linvi/tweetinvi/wiki/Exception-Handling#per-request-exception-handling) for more info.

**Comment by JosephBalzanoDev** [comment]
This is the code: 
```C#
var appCreds = new ConsumerCredentials(this.ConsumerKey, this.SecretKey);
if (appCreds != null)
{
      ret = AuthFlow.InitAuthentication(appCreds, this.RedirectUrl.ToString())?.AuthorizationURL ?? "";
} 
```
I tried it with two key pairs and the result is the same, ONLY on Windows Phone 8.1

**Comment by linvi** [comment]
> ONLY on Windows Phone 8.1

**It does not work** only on wp 8.1?

Surely the `if (appCreds != null)` is not necessary here as you are creating a new `ConsumerCredentials` using the `new` keyword.

**Comment by linvi** [comment]
If this the problem is on WP 8.1 please share your exception as explained here https://github.com/linvi/tweetinvi/wiki/Exception-Handling#per-request-exception-handling.

**Comment by JosephBalzanoDev** [comment]
Yes, it doesn't work only on Windows Phone 8.1 because on Windows Phone 10 it works properly. 
I don't have any exception from call but i recive `null` from this call:
`AuthFlow.InitAuthentication(appCreds, this.RedirectUrl.ToString())`.

**Comment by linvi** [comment]
Yes, this is because tweetinvi is by default configured to catch the errors for you. Please read https://github.com/linvi/tweetinvi/wiki/Exception-Handling#per-request-exception-handling.

**Comment by JosephBalzanoDev** [comment]
Sorry, I don't have read the property that disable the exception handling. 
I'm trying now.

**Comment by JosephBalzanoDev** [comment]
Okey, I have the exception from that call:
```
Eccezione generata: 'Tweetinvi.Core.Exceptions.TwitterException' in Tweetinvi.Credentials.ni.DLL
--- Date : 24/02/2016 15:58:35
URL : https://api.twitter.com/oauth/request_token
Code : 401
Error documentation description : Unauthorized -  Authentication credentials were missing or incorrect.
Error message : https://api.twitter.com/oauth/request_token web request failed.
Timestamp out of bounds. (135)
```
From the generated exception i understand that the pair of keys is incorrect because i received the unauthorized 401 error, but this it seems very strange because on Windows Phone 10 i don't have any exception.

**Comment by linvi** [comment]
Have you verified that the time on the Windows Phone is actually correct?

**Comment by JosephBalzanoDev** [comment]
To develop this application I need incorrect date and time but now I'm trying with the correct date and time.

**Comment by JosephBalzanoDev** [comment]
The result is the same: 
```
--- Date : 24/02/2016 10:45:31
URL : https://api.twitter.com/oauth/request_token
Code : 401
Error documentation description : Unauthorized -  Authentication credentials were missing or incorrect.
Error message : https://api.twitter.com/oauth/request_token web request failed.
Timestamp out of bounds. (135)
```

**Comment by linvi** [comment]
`Timestamp out of bounds. (135)` means that your clock is not correct. Either the Date or Time or Timezone is not correct here.

So you need to know that to sign the query for Twitter we need the correct DateTime. If not Twitter will reject the query.

I understand that there are some devices that might not have the correct date. Therefore I created a new Task that will give you the ability to set a lambda expression that you will be able to set to actually define how to get the current DateTime.

**Comment by JosephBalzanoDev** [comment]
Okay, with the correct date and time it works properly.

Really thank you so much for your great work and for the help! :)

**Comment by linvi** [comment]
I am happy that I could help.

Do you need the ability to set a lambda for DateTime in specific amount of time or am I good to do that as part of release 0.9.14.0?

**Comment by linvi** [comment]
And don't forget to star Tweetinvi ;)

**Comment by JosephBalzanoDev** [comment]
It is not a priority right now, but I would overjoyed to see her in the next release :)

**Comment by linvi** [comment]
It will be!

**Comment by linvi** [comment]
Just to let you know, the ability to set the DateTime is now available if you download the source code of 0.9.14.0.

**Comment by JosephBalzanoDev** [comment]
Oh great info, thank you for your work!
